modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -318,7 +323,20 @@ protected void setError(String executionId, String error) {
     }
 
     public String getError(String executionId) {
-        return (String) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR);
+        // Error on the operation/automation call, see #setError()
+        String error = (String) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR);
+        if (StringUtils.isNotEmpty(error)) {
+            return error;
+        }
+
+        if (!isAsync(executionId)) {
+            return null;
+        }
+
+        // Error on the bulk action
+        Serializable taskId = getTaskId(executionId);
+        BulkStatus status = Framework.getService(BulkService.class).getStatus((String) taskId);
```

Comment:
Relying directly on `BulkStatus` here breaks the `AsyncStatus` abstraction. We should introduce a `getError()` on `AsynStatus` instead and rely on `getAsyncService(executionId).getStatus(taskId).getError()`.

Reply:
Yep, I see you point, but I am wondering about the backward compatibility, If we do, I thing we have to make something like:

```

public interface AsyncStatus<K extends Serializable> extends Serializable {

    /**
     * Returns the number of errors encountered.
     */
    public default long getErrorCount() {
        return 0;
    }

    /**
     * Returns the first error message if any or null.
     */
    public default String getErrorMessage() {
        return null;
    }

    /**
     * Returns the number of errors encountered.
     */
    public default long getErrorCount() {
        return 0;
    }

    /**
     * Returns the first error message if any or null.
     */
    public default String getErrorMessage() {
        return null;
    }
....
```

Do you agree on this approche 

thx

Reply:
You mean in terms of possible implementations that would be missing this new method ? I'm ok with the default interface imlpementations as you proposed. As for the methods themselves up to you really, I'd introduce only the methods you'll actually use, having an error count without a way to access the other errors seems pretty pointless to me tbh. So, if in the end you only use one error message, which is the first error in the bulkstatus, this is an implementation detail imho and I'd have a simple `getError()` in `AsyncStatus` returning it

Reply:
> You mean in terms of possible implementations that would be missing this new method ? 

exactly, sorry I didn't mention it, but yes

> I'm ok with the default interface implementations as you proposed.

yep

> I'd introduce only the methods you'll actually use, having an error count without a way to access the other errors 

seems pretty pointless to me tbh. So, if in the end you only use one error message, which is the first error in the bulkstatus, this is an implementation detail imho and I'd have a simple getError() in AsyncStatus returning it
you are right, completely agree with you, I'll only keep `getError` one, sorry I don't I added the two ones

thx